### PR TITLE
add possibility to override default html templates

### LINKED ISSUE #45: Helm charts-  provide custom templates for error and token pages
**Is your feature request related to a problem? Please describe.**
Unable to provide/override custom html templates for loginapp

**Describe the solution you'd like**
Add configmap with `error.html` and `token.html` keys that will be mounted as a volume at `/web/templates/` overriding default templates.

**Describe alternatives you've considered**
Not much alternatives as this requires changes in deployment manifest

**Additional context**
Would be nice to be able to provide custom values that would be passed as is to a html templates (e.g. provide values in loginapp configuration)

### PR DESCRIPTION
**Pull request by Imunhatep**
Something similar to that might close #45

### CODE REVIEW

**Comment by justusbunsi** [comment]
After playing around to get this partial override implemented, I don't see any advantage in it anymore. 😀
At the moment there are only two template files so I guess, it's fine to override both files at once.

**Comment by fydrah** [comment]
Hi @Imunhatep ,

Thanks for this PR.

I tested the override, and I don't this this work as expected. If we want to fill a multi-line html template, this will not work. We must change configmap template with something like:

```
[...]
  error.html: |
    {{ .Values.templates.error | nindent 4 }}
[...]
```
Same for the example:
```
# Configuration overrides html templates content used by loginapp to display error and token pages
# Example:
#
#  templates:
#    token: |
#      <html> ....
templates:
```

Also, as @justusbunsi suggested, a partial override should be allowed. Maybe users will only want to override error or token template (mainly only the token template). It's only a few "if" conditions to add, and change "and" operators by "or". This is a few updates for a lot of more conveniences

File: helm/loginapp/templates/configmap.yaml [hunk 1]
```diff
@@ -7,4 +7,16 @@ metadata:
     {{- include "loginapp.labels" . | nindent 4 }}
 data:
   config.yaml: |
-{{ toYaml (mergeOverwrite $config .Values.configOverwrites) | nindent 4 }}
\ No newline at end of file
+{{ toYaml (mergeOverwrite $config .Values.configOverwrites) | nindent 4 }}
+---
+{{- if and .Values.templates.error .Values.templates.token}}
```
Comment (fydrah) [inline helm/loginapp/templates/configmap.yaml:12]: `or`

File: helm/loginapp/templates/configmap.yaml [hunk 2]
```diff
@@ -7,4 +7,16 @@ metadata:
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "loginapp.fullname" . }}-templates
+  labels:
+    {{- include "loginapp.labels" . | nindent 4 }}
+data:
```
Comment (fydrah) [inline helm/loginapp/templates/configmap.yaml:19]: * conditional include, or a loop over all `templates` keys
* should support multi-line (`|-` & `| nindent 4`)

File: helm/loginapp/templates/deployment.yaml [hunk 3]
```diff
@@ -69,6 +69,10 @@ spec:
           volumeMounts:
             - name: config
               mountPath: "/cfg"
+            {{- if and .Values.templates.error .Values.templates.token }}
```
Comment (fydrah) [inline helm/loginapp/templates/deployment.yaml:72]: `or`

File: helm/loginapp/templates/deployment.yaml [hunk 4]
```diff
@@ -81,6 +85,16 @@ spec:
       - name: config
         configMap:
           name: {{ include "loginapp.fullname" . }}
+      {{- if and .Values.templates.error .Values.templates.token }}
```
Comment (fydrah) [inline helm/loginapp/templates/deployment.yaml:88]: `or`

File: helm/loginapp/templates/deployment.yaml [hunk 5]
```diff
@@ -81,6 +85,16 @@ spec:
...
+      - name: templates
+        configMap:
+          name: {{ include "loginapp.fullname" . }}-templates
+          items:
```
Comment (fydrah) [inline helm/loginapp/templates/deployment.yaml:92]: * conditional include, or a loop over all `templates` keys
* should support multi-line (`|-` & `| nindent 4`)

File: helm/loginapp/values.yaml [hunk 6]
```diff
@@ -152,6 +152,16 @@ config:
 #    scopes: [openid,profile,email]
 configOverwrites: {}
 
+
+# Configuration overrides html templates content used by loginapp to display error and token pages
+# Example:
+#
+#  templates:
+#    token: {{`<html><body><title>Hello token</title></body></html>`}}
```
Comment (fydrah) [inline helm/loginapp/values.yaml:160]: update example with multi-line

**Review by fydrah — CHANGES_REQUESTED** [review]

**Comment by Imunhatep** [comment]
Personally I do not consider " a partial override " as a must have. 
Would prefer to leave this part for a separate PR to the ones who would ask/suggest it.

**Comment by fydrah** [comment]
Thanks @Imunhatep  for the update, I tested the new version and it works pretty well with the loop over templates names :)

You're right it's not a must have, but it's an improvement!

Also thanks for the dex dependency update, it has been updated last time (by mistake..) to this old repository, and not the  official one..

**Pull request merged**